### PR TITLE
Support msi

### LINKED ISSUE #98: Feature request: Support MSI when running in a VM or Azure Cloud Shell
Now that Chef Workstation includes Test Kitchen and the Azure driver when running on Azure Cloud Shell, it would be good if we could use an MSI instead of having to provide a service principal in ~/.azure/credentials

### PR DESCRIPTION
**Pull request by hantonov**
### Description

Support managed identity (MSI) when running on Azure VM.
No need for setting up credentials file.  

### Issues Resolved

Resolves #98 

Signed-off-by: Hillel Antonov <hantonov@gmail.com>

### CODE REVIEW

File: lib/kitchen/driver/azure_credentials.rb [hunk 1]
```diff
@@ -28,7 +28,7 @@ def initialize(subscription_id:, environment: "Azure")
         if File.file?(config_file)
           @credentials = IniFile.load(File.expand_path(config_file))
         else
-          warn "#{CONFIG_PATH} was not found or not accessible."
+          warn "#{CONFIG_PATH} was not found or not accessible. Will use environemnt vriables or MSI"
```
Comment (ehanlon) [inline lib/kitchen/driver/azure_credentials.rb:31]:
```suggestion
          warn "#{CONFIG_PATH} was not found or not accessible. Will use environment variables or MSI"
```

File: lib/kitchen/driver/azure_credentials.rb [hunk 2]
```diff
@@ -38,33 +38,49 @@ def initialize(subscription_id:, environment: "Azure")
...
+      def credentials
+        @credentials ||= {}
+      end
+
+      def credentials_property(property)
+        credentials[subscription_id]&.[](property)
+      end
+
+      def tenant_id!
+        tenant_id || raise("Must provide tenant id. Use AZURE_TENANT_ID environment variable or set it in crednetials file")
```
Comment (ehanlon) [inline lib/kitchen/driver/azure_credentials.rb:61]:
```suggestion
        tenant_id || raise("Must provide tenant id. Use AZURE_TENANT_ID environment variable or set it in credentials file")
```

File: lib/kitchen/driver/azure_credentials.rb [hunk 3]
```diff
@@ -38,33 +38,49 @@ def initialize(subscription_id:, environment: "Azure")
...
                     credentials: ::MsRest::TokenCredentials.new(token_provider),
                     active_directory_settings: ad_settings,
                     base_url: endpoint_settings.resource_manager_endpoint_url }
-
+        options[:client_id] = client_id if client_id
+        options[:client_secret] = client_secret if client_secret
         options
       end
 
       private
```
Comment (ehanlon) [inline lib/kitchen/driver/azure_credentials.rb:51]:
```suggestion
      private
      
```
There should be a space before and after `private`.

**Comment by tas50** [comment]
MSI support merged in #134 and released yesterday!

**Pull request closed without merging**